src/handle_provide_dynamic_descriptor.c

```diff
@@ -268,50 +141,38 @@ static int save_dynamic_token_info(const tlv_extracted_t *tlv_extracted,
     PRINTF("magnitude            = %d\n", g_dynamic_token_info.magnitude);
     PRINTF("encoded_mint_address = %s\n", g_dynamic_token_info.encoded_mint_address);
     PRINTF("mint_address         = %.*H\n", PUBKEY_LENGTH, g_dynamic_token_info.mint_address);
+
     return 0;
 }
 
-void handle_provide_dynamic_descriptor(void) {
-    trusted_info_reset(&g_dynamic_token_info);
-    // Main structure that will received the parsed TLV data
-    tlv_extracted_t tlv_extracted = {0};
+dynamic_token_info_t g_dynamic_token_info;
 
-    PRINTF("Received trusted info, length = %d\n", G_command.message_length);
+void handle_provide_dynamic_descriptor(void) {
+    explicit_bzero(&g_dynamic_token_info, sizeof(g_dynamic_token_info));
 
-    // The parser will fill it with the hash of the whole TLV payload (except SIGN tag)
-    cx_sha256_init(&tlv_extracted.hash_ctx);
+    tlv_dynamic_descriptor_out_t tlv_output = {0};
 
     // Convert G_command to buffer_t format. 0 copy
     buffer_t payload = {.ptr = G_command.message, .size = G_command.message_length};
 
-    // Call the function created by the macro from the TLV lib
-    if (!parse_dynamic_token_tag(&payload, &tlv_extracted, &tlv_extracted.received_tags)) {
-        PRINTF("Failed to parse tlv payload\n");
+    if (tlv_use_case_parse_dynamic_descriptor_payload(&payload, &tlv_output) != 0) {
+        PRINTF("tlv_use_case_parse_dynamic_descriptor_payload failed\n");
         THROW(ApduReplySolanaInvalidDynamicToken);
     }
 
-    // Verify that the fields received are correct in our context
-    if (verify_struct(&tlv_extracted) != 0) {
-        PRINTF("Failed to verify tlv payload\n");
+    tlv_TUID_data_t tlv_TUID_data;
+    PRINTF("tlv_output.TUID %.*H\n", tlv_output.TUID.size, tlv_output.TUID.ptr);
+    if (!parse_dynamic_token_tuid(&tlv_output.TUID, &tlv_TUID_data, &tlv_TUID_data.received_tags)) {
```

Comment:
Couldn't we make this operation in the `tlv_use_case_parse_dynamic_descriptor_payload()` 🤔 ?
This would avoid to still use the `DEFINE_TLV_PARSER` macro in this handler ?